### ISSUE TITLE
Please upgrade libwagon to version 3.0.0

### ISSUE THREAD

**Issue by ehashman** [technomancy/leiningen#2443]
As reported in the dependency https://github.com/cemerick/pomegranate/issues/101, upstream maven is using wagon 3.0.0 and version 2.12 is no longer supported.

Hence, we should upgrade when possible.

**Comment by ehashman** [comment]
(Note: Leiningen cannot build properly in Debian without bumping this, as Maven has been upgraded to 3.5.3 which is pulling in the new, conflicting libwagon.)

**Comment by ehashman** [comment]
@winks so good news, turns out this is not blocking Debian (at least from a build perspective). I submitted https://github.com/cemerick/pomegranate/issues/101 and corresponding PR https://github.com/cemerick/pomegranate/pull/103 which passed all tests.

Because Leiningen was failing to build from source in Debian, I had to patch all this stuff to get it working. It hasn't hit testing yet so I can't say for certain but the patched lein seems to be working okay. I think it's pretty safe to proceed with this, the major version bump in wagon didn't appear to be breaking.
